src/Network/Affjax/Request.purs

```diff
@@ -0,0 +1,115 @@
+module Network.Affjax.Request
+  ( Ajax()
+  , AjaxRequest()
+  , MethodName(..)
+  , Content(..)
+  , AjaxResponse()
+  , defaultRequest
+  , ajax
+  ) where
+
+import Control.Monad.Aff (Aff(), EffA(), makeAff)
+import Control.Monad.Eff (Eff())
+import Control.Monad.Eff.Exception(Error())
+import Data.Array ()
+import Data.ArrayBuffer.Types (ArrayView())
+import Data.Function (Fn8(), runFn8)
+import Data.Maybe (Maybe(..), maybe)
+import Data.Nullable (Nullable(), toNullable)
+import DOM (Document())
+import DOM.File (Blob())
+import DOM.XHR (FormData())
+import Network.HTTP (Header(..))
+
+-- | The event type for AJAX requests.
+foreign import data Ajax :: !
+
+-- | The parameters for an AJAX request.
+type AjaxRequest a =
+  { url :: String
+  , method :: MethodName
+  , headers :: [Header]
+  , content :: Maybe (Content a)
+  , username :: Maybe String
+  , password :: Maybe String
+  }
+
+-- | A HTTP method name: `GET`, `POST`, etc.
+newtype MethodName = MethodName String
```

Comment:
1. `StatusCode` does not support custom.
2. `Verb` does not support custom.
3. `Header` is not well-typed; the value is a `String` but much more precise types are possible (values have a lot of structure, they are almost never free-form text!).
4. `Show` is abused; `Show` instances should render as REPL-friendly copy/paste code. Instead, functions should be used for rendering these things.
5. Not all known verbs, codes, header heads are implemented.

I do think we need a good http-like library, one which is type safe and maintained. That may be too big a project for now (but open to whatever your thoughts are on the issue).